### PR TITLE
Fix factorify and createInstance/build

### PR DESCRIPTION
**Pull request by lundibundi**
* Support short domain notation in factorify e.g. { field: 'domain' }
* don't dismiss falsy values in factorify as invalid
* add tests for Logical field and Logical in schema

And
* Make createInstance check required fields (Refs: https://github.com/metarhia/metaschema/pull/111#discussion_r224598559)

Fixes: https://github.com/metarhia/metaschema/issues/120

This also adds `metatests` though only for the newly added tests as there is already https://github.com/metarhia/metaschema/pull/111

### CODE REVIEW

**Comment by lundibundi** [comment]
Because it's related to the other fix and I didn't want to add metatests (and schema test) in the other separate PR again as that would certainly cause conflicts for no reason.

File: lib/schema.js [hunk 1]
```diff
@@ -186,12 +187,17 @@ const factorify = (
...
-        type = def.category || def.domain;
+        type = def.category || def.domain || def;
         value = createInstance(type, value);
-        if (value) obj[field] = value;
+        if (value !== null) obj[field] = value;
         else return null;
       }
     }
+    for (const prop of required) {
+      if (obj[prop] === undefined) {
```
Comment (belochub) [inline lib/schema.js]: `null` should also be prohibited.

**Comment by lundibundi** [comment]
ping @belochub @tshemsedinov

File: test/schema.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const { testSync } = require('metatests');
+const ms = require('../');
```
Comment (tshemsedinov) [inline test/schema.js]: better `metaschema`

File: lib/schema.js [hunk 3]
```diff
@@ -186,12 +187,17 @@ const factorify = (
         value = def(value);
         obj[field] = value;
       } else {
-        type = def.category || def.domain;
+        type = def.category || def.domain || def;
```
[the remark this answers is not in the document]
Comment (lundibundi) [reply lib/schema.js]: For some reason, I remember having just domain name string as value in schemas.
@tshemsedinov could you clarify if such schema should be allowed
```js
{
  field: 'domainName'
}
```
OR only
```js
{
  field: { domain: 'domainName' }
}
```
is allowed?

File: test/schema.js (hunk 2 above)
Comment (lundibundi) [reply test/schema.js]: Ok

File: lib/schema.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lundibundi) [reply lib/schema.js]: ping @tshemsedinov

[the remark this answers is not in the document]
Comment (tshemsedinov) [reply lib/schema.js]: Just following is acceptable schema syntax:
```js
{
  field: { domain: 'domainName' }
}
```

**Comment by lundibundi** [comment]
Landed in a8e45c6e6e1314f6dec671d7d1ef93bc4e6f1974 and 410c2798bee6c099fabcf2a9d0a7ee3e3fffe9d2.

**Pull request merged**